ray-curator/ray_curator/models/cosmos_embed1.py

```diff
@@ -0,0 +1,187 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Model Cosmos-Embed1."""
+
+from pathlib import Path
+from typing import Final, Literal, cast
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from loguru import logger
+from transformers import AutoModel, AutoProcessor
+
+from .base import ModelInterface
+
+_COSMOS_EMBED1_VARIANTS_INFO: Final = {
+    "224p": "nvidia/Cosmos-Embed1-224p",
+    "336p": "nvidia/Cosmos-Embed1-336p",
+    "448p": "nvidia/Cosmos-Embed1-448p",
+}
+
+
+class CosmosEmbed1(ModelInterface):
+    """Cosmos-Embed1 embedding model."""
+
+    def __init__(self, *, variant: Literal["224p", "336p", "448p"] = "336p", utils_only: bool = False, model_dir: str | None = None) -> None:
+        """Initialize Cosmos-Embed1 model.
+
+        Args:
+            variant: Choose from "224p", "336p", "448p".
+            utils_only: Whether to only initialize utility functions.
+
+        """
+        super().__init__()
+        self.variant = variant
+        self._weights_name = _COSMOS_EMBED1_VARIANTS_INFO[variant]
+        self._weights_dir = str(Path(model_dir) / self._weights_name)
+        self._utils_only = utils_only
+        self._model: AutoModel | None = None
+
+    @property
+    def conda_env_name(self) -> str:
+        """Get the conda environment name.
+
+        Returns:
+            The conda environment name.
+
+        """
+        return "unified"
+
+    @property
+    def model_id_names(self) -> list[str]:
+        """Get the model ID names.
+
+        Returns:
+            A list of model ID names.
+
+        """
+        return [self._weights_name]
+
+    def setup(self) -> None:
+        """Set up the Cosmos-Embed1 model.
+
+        This method initializes the model and its configuration for processing video and text data.
+        """
+        logger.info("Setting up Cosmos-Embed1 model")
+        if not Path(self._weights_dir).exists():
```

Comment:
Just wondering, can't we use `setup on node` to make this directory and download these weights?